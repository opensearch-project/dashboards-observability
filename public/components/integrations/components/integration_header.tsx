/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import {
  EuiButton,
  EuiContextMenuItem,
  EuiContextMenuPanel,
  EuiLink,
  EuiPageHeader,
  EuiPageHeaderSection,
  EuiPopover,
  EuiSpacer,
  EuiTab,
  EuiTabs,
  EuiText,
  EuiTitle,
} from '@elastic/eui';
import React, { useState } from 'react';
import {
  OPENSEARCH_CATALOG_URL,
  OPENSEARCH_DOCUMENTATION_URL,
} from '../../../../common/constants/integrations';
import { IntegrationUploadFlyout } from './upload_flyout';

export const IntegrationHeaderActions = ({ onShowUpload }: { onShowUpload: () => void }) => {
  const [isPopoverOpen, setPopover] = useState(false);

  const closePopover = () => {
    setPopover(false);
  };

  const onButtonClick = () => {
    setPopover((isOpen) => !isOpen);
  };

  const items = [
    <EuiContextMenuItem
      onClick={() => {
        closePopover(); // If the popover isn't closed, it overlays over the flyout
        onShowUpload();
      }}
    >
      Upload Integrations
    </EuiContextMenuItem>,
    <EuiContextMenuItem href={OPENSEARCH_CATALOG_URL}>View Catalog</EuiContextMenuItem>,
  ];
  const button = (
    <EuiButton iconType="arrowDown" fill={true} onClick={onButtonClick}>
      Catalog
    </EuiButton>
  );
  return (
    <EuiPopover
      id="integHeaderActionsPanel"
      button={button}
      isOpen={isPopoverOpen}
      closePopover={closePopover}
      panelPaddingSize="none"
      anchorPosition="downLeft"
    >
      <EuiContextMenuPanel items={items} />
    </EuiPopover>
  );
};

export const IntegrationHeader = () => {
  const tabs = [
    {
      id: 'installed',
      name: 'Installed',
      disabled: false,
    },
    {
      id: 'available',
      name: 'Available',
      disabled: false,
    },
  ];

  const [selectedTabId, setSelectedTabId] = useState(
    window.location.hash.substring(2) ? window.location.hash.substring(2) : 'installed'
  );
  const [showUploadFlyout, setShowUploadFlyout] = useState(false);

  const onSelectedTabChanged = (id: string) => {
    setSelectedTabId(id);
    window.location.hash = id;
  };

  const renderTabs = () => {
    return tabs.map((tab, index) => (
      <EuiTab
        onClick={() => onSelectedTabChanged(tab.id)}
        isSelected={tab.id === selectedTabId}
        disabled={tab.disabled}
        key={index}
      >
        {tab.name}
      </EuiTab>
    ));
  };

  return (
    <div>
      <EuiPageHeader>
        <EuiPageHeaderSection>
          <EuiTitle size="l" data-test-subj="integrations-header">
            <h1>Integrations</h1>
          </EuiTitle>
        </EuiPageHeaderSection>
        <EuiPageHeaderSection>
          <IntegrationHeaderActions onShowUpload={() => setShowUploadFlyout(true)} />
        </EuiPageHeaderSection>
      </EuiPageHeader>
      <EuiSpacer size="s" />
      <EuiText size="s" color="subdued">
        View integrations with preconfigured assets immediately within your OpenSearch setup.{' '}
        <EuiLink external={true} href={OPENSEARCH_DOCUMENTATION_URL} target="blank">
          Learn more
        </EuiLink>
      </EuiText>
      <EuiSpacer size="l" />
      <EuiTabs display="condensed">{renderTabs()}</EuiTabs>
      <EuiSpacer size="s" />
      {showUploadFlyout ? (
        <IntegrationUploadFlyout onClose={() => setShowUploadFlyout(false)} />
      ) : null}
    </div>
  );
};
