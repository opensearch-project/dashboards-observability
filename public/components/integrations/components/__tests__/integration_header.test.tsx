/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { waitFor } from '@testing-library/react';
import { IntegrationHeader } from '../integration_header';

describe('Integration Header Test', () => {
  configure({ adapter: new Adapter() });

  it('Renders integration header as expected', async () => {
    const wrapper = mount(<IntegrationHeader />);

    await waitFor(() => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
