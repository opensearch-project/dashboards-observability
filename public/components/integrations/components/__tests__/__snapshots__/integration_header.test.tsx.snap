// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integration Header Test Renders integration header as expected 1`] = `
<IntegrationHeader>
  <div>
    <EuiPageHeader>
      <header
        className="euiPageHeader euiPageHeader--responsive euiPageHeader--center"
      >
        <EuiPageHeaderSection>
          <div
            className="euiPageHeaderSection"
          >
            <EuiTitle
              data-test-subj="integrations-header"
              size="l"
            >
              <h1
                className="euiTitle euiTitle--large"
                data-test-subj="integrations-header"
              >
                Integrations
              </h1>
            </EuiTitle>
          </div>
        </EuiPageHeaderSection>
      </header>
    </EuiPageHeader>
    <EuiSpacer
      size="s"
    >
      <div
        className="euiSpacer euiSpacer--s"
      />
    </EuiSpacer>
    <EuiText
      color="subdued"
      size="s"
    >
      <div
        className="euiText euiText--small"
      >
        <EuiTextColor
          color="subdued"
          component="div"
        >
          <div
            className="euiTextColor euiTextColor--subdued"
          >
            View integrations with preconfigured assets immediately within your OpenSearch setup.
             
            <EuiLink
              external={true}
              href="https://opensearch.org/docs/latest/integrations/index"
              target="blank"
            >
              <a
                className="euiLink euiLink--primary"
                href="https://opensearch.org/docs/latest/integrations/index"
                rel="noreferrer"
                target="blank"
              >
                Learn more
                <EuiIcon
                  aria-label="External link"
                  className="euiLink__externalIcon"
                  size="s"
                  type="popout"
                >
                  <EuiIconBeaker
                    aria-hidden={true}
                    aria-label="External link"
                    className="euiIcon euiIcon--small euiIcon-isLoading euiLink__externalIcon"
                    focusable="false"
                    role="img"
                    style={null}
                  >
                    <svg
                      aria-hidden={true}
                      aria-label="External link"
                      className="euiIcon euiIcon--small euiIcon-isLoading euiLink__externalIcon"
                      focusable="false"
                      height={16}
                      role="img"
                      style={null}
                      viewBox="0 0 16 16"
                      width={16}
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
                      />
                    </svg>
                  </EuiIconBeaker>
                </EuiIcon>
              </a>
            </EuiLink>
          </div>
        </EuiTextColor>
      </div>
    </EuiText>
    <EuiSpacer
      size="l"
    >
      <div
        className="euiSpacer euiSpacer--l"
      />
    </EuiSpacer>
    <EuiTabs
      display="condensed"
    >
      <div
        className="euiTabs euiTabs--condensed"
        role="tablist"
      >
        <EuiTab
          disabled={false}
          isSelected={true}
          key="0"
          onClick={[Function]}
        >
          <button
            aria-selected={true}
            className="euiTab euiTab-isSelected"
            disabled={false}
            onClick={[Function]}
            role="tab"
            type="button"
          >
            <span
              className="euiTab__content"
            >
              Installed
            </span>
          </button>
        </EuiTab>
        <EuiTab
          disabled={false}
          isSelected={false}
          key="1"
          onClick={[Function]}
        >
          <button
            aria-selected={false}
            className="euiTab"
            disabled={false}
            onClick={[Function]}
            role="tab"
            type="button"
          >
            <span
              className="euiTab__content"
            >
              Available
            </span>
          </button>
        </EuiTab>
      </div>
    </EuiTabs>
    <EuiSpacer
      size="s"
    >
      <div
        className="euiSpacer euiSpacer--s"
      />
    </EuiSpacer>
  </div>
</IntegrationHeader>
`;
